source/parse.h

```diff
@@ -519,6 +519,7 @@ struct type_id_node
 {
     source_position pos;
 
+    token const* static_qualifier = {};
```

Comment:
[Explanation] I could have easily included `static` in `pc_qualifiers` (that could be renamed). I decided against that as `static` is of somewhat different nature.

Reply:
Yes, that's the way I tend to handle these qualifier.

But I feel bad that I'm going to close this PR unmerged now, seeing all the work you've put into it -- as I mentioned on #212 I'm still thinking about what I want to do for `static` variables and I'd like to push them down the road, especially to get experience with classes first (and how the demands for static members unfolds there)... sorry, and thanks for understanding!